sdk/core/core/internal/az_span_internal.h

```diff
@@ -37,6 +37,10 @@ AZ_INLINE AZ_NODISCARD int32_t _az_span_diff(az_span sliced_span, az_span origin
  *         - #AZ_OK if successful
  *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the \p destination is not big enough to contain
  * the encoded bytes
+ *
+ * @remark If \p destination can't fit the \p source, some data may still be written to it, but the
+ * \p out_length will be set to 0, and the function will return #AZ_ERROR_INSUFFICIENT_SPAN_SIZE.
+ * The \p destination and \p source must not overlap.
```

Comment:
Btw, how does this render in docs? Do we need to split this into its own `@remarks` line?

Reply:
"Multiple adjacent \remark commands will be joined into a single paragraph. Each remark will start on a new line." http://www.doxygen.nl/manual/commands.html#cmdremark